File: src/forms/form_textareas.py
"""
Extraction utilities for textarea controls.
"""
from forms.utils.form_extract_common import find_label_for

def extract_textareas(container):
    """Extract textarea elements"""
    textareas = container.find_all('textarea')
    
    result = []
    for textarea in textareas:
        textarea_info = {
            'id': textarea.get('id', ''),
            'name': textarea.get('name', ''),
            'value': textarea.text,
            'required': textarea.has_attr('required'),
            'placeholder': textarea.get('placeholder', ''),
            'rows': textarea.get('rows', ''),
            'cols': textarea.get('cols', ''),
            'disabled': textarea.has_attr('disabled'),
            'readonly': textarea.has_attr('readonly')
        }
        
        # Try to find label
        textarea_info['label'] = find_label_for(container, textarea)
        
        result.append(textarea_info)
        
    return result

File: js-src/forms/form_textareas.js
/**
 * Extraction utilities for textarea controls.
 */
const { findLabelFor } = require('./form_common');

/**
 * Extract textarea elements
 * @param {Object} $ - Cheerio instance
 * @param {Object} container - Container element
 * @returns {Array} Array of textarea objects
 */
function extractTextareas($, container) {
  const textareas = [];
  
  $(container).find('textarea').each((i, textarea) => {
    const textareaInfo = {
      id: $(textarea).attr('id') || '',
      name: $(textarea).attr('name') || '',
      value: $(textarea).text(),
      required: $(textarea).attr('required') !== undefined,
      placeholder: $(textarea).attr('placeholder') || '',
      rows: $(textarea).attr('rows') || '',
      cols: $(textarea).attr('cols') || '',
      disabled: $(textarea).attr('disabled') !== undefined,
      readonly: $(textarea).attr('readonly') !== undefined
    };
    
    // Try to find label
    textareaInfo.label = findLabelFor($, textarea);
    
    textareas.push(textareaInfo);
  });
  
  return textareas;
}

module.exports = { extractTextareas };
